string/manipulate.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @typedef {{
+ *      string: string,
+ * }|Object} mojave.replacementStringObject
+ */
+
+
+/**
+ * Replaces all occurrences of a given matching string with a given replacement string
```

Comment:
> Replaces all occurrences of substrings in the given string.

Reply:
I agree, the current description is a bit complicated.